sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1700,6 +1700,156 @@ object functions {
    */
   def length(columnName: String): Column = length(Column(columnName))
 
+  //////////////////////////////////////////////////////////////////////////////////////////////
+  // DateTime functions
+  //////////////////////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * Converts a date/timestamp/string to a value of string in the format specified by the date
+   * format given by the second argument
+   *
+   * NOTE: Prefer using functions like year. These use an optimized implementation.
+   * @group datetime_funcs
+   * @since 1.5.0
+   */
+  def dateFormat(l: Column, r: String): Column = DateFormatClass(l.expr, Literal(r))
```

Comment:
use date_format; we've decided to follow sql function names instead of java style.